### PR TITLE
[OTX] Drop legacy-resolver

### PR DESCRIPTION
**Pull request by cih9088**
### Changes
- Since pip version is not restricted to `21.2.1` anymore, we can drop `legacy-resolver`. This slow version resolve issue has been fixed in version of `22.0`.

[Release note of pip 22.0](https://pip.pypa.io/en/stable/news/#v22-0)
> Optimize installation order calculation to improve performance when installing requirements that form a complex dependency graph with a large amount of edges. ([#10557](https://github.com/pypa/pip/issues/10557))

### CODE REVIEW

**Review by harimkang — APPROVED** [review]
LGTM! I'll test it in my environment.

**Review by sungmanc — APPROVED** [review]
Thanks for this PR

**Review by goodsong81 — APPROVED** [review]
Nice.

**Comment by cih9088** [comment]
@harimkang you might encounter timm version miss match cuased by `otxreid`, but as far as I know this will be deprecated so that you could safely ignore it.

**Comment by harimkang** [comment]
> @harimkang you might encounter timm version miss match cuased by `otxreid`, but as far as I know this will be deprecated so that you could safely ignore it.

Oh, yes i checked. However, this script doesn't work in that conflict and stops the installation, is that okay?
image

How to ignore..?

**Comment by cih9088** [comment]
> Oh, yes i checked. However, this script doesn't work in that conflict and stops the installation, is that okay?
> How to ignore..?

@harimkang 
Just comment out `otxreid` in `requirements/classification.txt`. In the end, it will be no use.

**Comment by harimkang** [comment]
> > Oh, yes i checked. However, this script doesn't work in that conflict and stops the installation, is that okay?
> > How to ignore..?
> 
> @harimkang Just comment out `otxreid` in `requirements/classification.txt`. In the end, it will be no use.

Yes, but there is a part where `torchreid` is still tied to classification training.
Shouldn't this PR be merged together as `torchreid` disappears later?

**Comment by cih9088** [comment]
> Yes, but is there a part where `torchreid` is still tied to training? Shouldn't this PR be merged together as `torchreid` disappears later?

The only dependency is NNCF task for classification and we could completely drop `torchreid` when https://github.com/openvinotoolkit/training_extensions/pull/1435 is merged into this branch.
Note that timm version mismatch does not introduced by this PR but it is just caused by in progress refactoring.

**Comment by goodsong81** [comment]
> > Yes, but is there a part where `torchreid` is still tied to training? Shouldn't this PR be merged together as `torchreid` disappears later?
> 
> The only dependency is NNCF task for classification and we could completely drop `torchreid` when #1435 is merged into this branch. Note that timm version mismatch does not introduced by this PR but it is just caused by in progress refactoring.

Yeah, let's merge this PR after that.

**Comment by cih9088** [comment]
@harimkang @sungmanc @goodsong81 Can we merge this as well since https://github.com/openvinotoolkit/training_extensions/pull/1497 was merged?

FYI, this PR also removes lots of version conflict errors such as following, thanks to new pip version resolver.
> ERROR: pip's legacy dependency resolver does not consider dependency conflicts when selecting packages. This behaviour is the source of the following dependency conflicts.
> tensorflow 2.9.3 requires protobuf<3.20,>=3.9.2, but you'll have protobuf 3.20.3 which is incompatible.
> tensorflow 2.9.3 requires tensorboard<2.10,>=2.9, but you'll have tensorboard 2.11.2 which is incompatible.
> nncf 2.3.0.dev0+9b20485 requires scipy<=1.9.1,>=1.3.2, but you'll have scipy 1.10.0 which is incompatible.
> openmodelzoo-modelapi 2022.3.0 requires numpy<=1.23.1,>=1.16.6, but you'll have numpy 1.23.4 which is incompatible.
> openmodelzoo-modelapi 2022.3.0 requires scipy<1.9.0,>=1.5.4, but you'll have scipy 1.10.0 which is incompatible.
> mmdeploy 0.10.0 requires protobuf<=3.20.1, but you'll have protobuf 3.20.3 which is incompatible.
> aiohttp 3.8.3 requires charset-normalizer<3.0,>=2.0, but you'll have charset-normalizer 3.0.1 which is incompatible.
> pytorch-lightning 1.6.5 requires protobuf<=3.20.1, but you'll have protobuf 3.20.3 which is incompatible.
> torchtext 0.12.0 requires torch==1.11.0, but you'll have torch 1.8.2+cu111 which is incompatible.
> anomalib 0.3.6 requires timm==0.5.4, but you'll have timm 0.6.12 which is incompatible.

w/o PR using cache
> ./otx/algorithms/init_venv.sh venv-test  184.73s user 27.87s system 68% cpu 5:08.20 total

w/ PR using cache
> ./otx/algorithms/init_venv.sh venv-test2  205.77s user 33.83s system 75% cpu 5:18.76 total

**Comment by sungmanc** [comment]
> @harimkang @sungmanc @goodsong81 Can we merge this as well since #1497 was merged?
> 
> FYI, this PR also removes lots of version conflict errors such as following, thanks to new pip version resolver.
> 
> > ERROR: pip's legacy dependency resolver does not consider dependency conflicts when selecting packages. This behaviour is the source of the following dependency conflicts.
> > tensorflow 2.9.3 requires protobuf<3.20,>=3.9.2, but you'll have protobuf 3.20.3 which is incompatible.
> > tensorflow 2.9.3 requires tensorboard<2.10,>=2.9, but you'll have tensorboard 2.11.2 which is incompatible.
> > nncf 2.3.0.dev0+9b20485 requires scipy<=1.9.1,>=1.3.2, but you'll have scipy 1.10.0 which is incompatible.
> > openmodelzoo-modelapi 2022.3.0 requires numpy<=1.23.1,>=1.16.6, but you'll have numpy 1.23.4 which is incompatible.
> > openmodelzoo-modelapi 2022.3.0 requires scipy<1.9.0,>=1.5.4, but you'll have scipy 1.10.0 which is incompatible.
> > mmdeploy 0.10.0 requires protobuf<=3.20.1, but you'll have protobuf 3.20.3 which is incompatible.
> > aiohttp 3.8.3 requires charset-normalizer<3.0,>=2.0, but you'll have charset-normalizer 3.0.1 which is incompatible.
> > pytorch-lightning 1.6.5 requires protobuf<=3.20.1, but you'll have protobuf 3.20.3 which is incompatible.
> > torchtext 0.12.0 requires torch==1.11.0, but you'll have torch 1.8.2+cu111 which is incompatible.
> > anomalib 0.3.6 requires timm==0.5.4, but you'll have timm 0.6.12 which is incompatible.

I want to make sure the current status,

After merging https://github.com/openvinotoolkit/training_extensions/pull/1497, don't we need to used torchreid anymore?
Then could you add the commit that comment out `otxreid`?

**Comment by cih9088** [comment]
> After merging #1497, don't we need to used torchreid anymore? Then could you add the commit that comment out `otxreid`?

I believe we do not need it as only dependency was NNCF task. NNCF for classification was mitigated to mmcls.
`otxreid` has been commented out already and all related codes have been removed in #1497.

**Comment by sungmanc** [comment]
> > After merging #1497, don't we need to used torchreid anymore? Then could you add the commit that comment out `otxreid`?
> 
> I believe we do not need it as only dependency was NNCF task. NNCF for classification was mitigated to mmcls. `otxreid` has been commented out already and removed in #1497.

Thanks for the information, then I don't have any objection for merging this.

**Review by sungmanc — APPROVED** [review]

**Pull request merged**
